### PR TITLE
NAS-130168 / 24.10 / NAS-130168

### PR DESCRIPTION
**Pull request by jnbastoky**
Add Glacier Instant Retrieval storage class for AWS S3 Cloud Sync

**Changes:**

Adds `Glacier Instant Retrieval` to the storage class option for AWS S3 Cloud Sync task.  `GLACIER_IR` was added to Rclone in v1.58.0: https://rclone.org/changelog/#v1-58-0-2022-03-18

**Testing:**

https://ixsystems.atlassian.net/browse/NAS-130168

### Downstream

|Affects         |Reasoning
|----------------|-------------------------------
|Documentation   | None found
|midddleware | Added to middleware in truenas/middleware#14094

### CODE REVIEW

File: src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts [hunk 1]
```diff
@@ -151,6 +151,7 @@ export class CloudSyncFormComponent implements OnInit {
     { label: 'One Zone-IA', value: 'ONEZONE_IA' },
     { label: 'Intelligent-Tiering', value: 'INTELLIGENT_TIERING' },
     { label: 'Glacier', value: 'GLACIER' },
+    { label: 'Glacier Instant Retrieval', value: 'GLACIER_IR' },
```
Comment (bvasilenko) [inline src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts:154]: Selection of this option causes validation error from middleware.

`cloud_sync_create.attributes.storage_class: Invalid choice: GLACIER_IR [EINVAL]`

Looks like it's not supported by middleware yet.

**Review by bvasilenko — CHANGES_REQUESTED** [review]

File: src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts (hunk 1 above)
Comment (jnbastoky) [reply src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts:154]: A PR #14094 has been submitted for middleware.  But it's not getting reviewed.  How do I get it reviewed?

Comment (undsoft) [reply src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts:154]: I've pinged some people to review.

Comment (undsoft) [reply src/app/pages/data-protection/cloudsync/cloudsync-form/cloudsync-form.component.ts:154]: Middleware PR was merged.

**Review by bvasilenko — APPROVED** [review]
Middleware is able to accept the new value of `storageClass` `GLACIER_IR`

**Pull request merged**